timeseries/src/autogluon/timeseries/models/tabular/tabular_model.py

```diff
@@ -0,0 +1,207 @@
+
+import numpy as np
+import pandas as pd
+from gluonts.time_feature import TimeFeature, get_lags_for_frequency, time_features_from_frequency_str
+from joblib import Parallel, delayed
+
+import autogluon.core as ag
+from autogluon.common.utils.log_utils import set_logger_verbosity
+from autogluon.tabular import TabularPredictor
+from autogluon.timeseries.dataset.ts_dataframe import ITEMID, TIMESTAMP, TimeSeriesDataFrame
+from autogluon.timeseries.models.abstract import AbstractTimeSeriesModel
+
+TS_METRIC_TO_TABULAR_METRIC = {
+    "mean_wQuantileLoss": "pinball_loss",
+    "MAPE": "mean_absolute_percentage_error",
+    "sMAPE": "mean_absolute_percentage_error",
+    "MASE": "mean_absolute_error",
+    "RMSE": "root_mean_squared_error",
+    "MSE": "mean_squared_error",
+}
+
+logger = logging.getLogger(__name__)
+
+
+class TabularModel(AbstractTimeSeriesModel):
+    """Uses TabularPredictor to forecast future time series values one step at a time.
+
+    The forecasting is converted to a tabular problem using the following features:
+
+    - lag features (observed time series values) based on ``freq`` of the data
+    - time features (e.g., day of the week) based on the timestamp of the measurement
+    """
+
+    def __init__(
+        self,
+        freq: Optional[str] = None,
+        prediction_length: int = 1,
+        path: Optional[str] = None,
+        name: Optional[str] = None,
+        eval_metric: str = None,
+        hyperparameters: Dict[str, Any] = None,
+        **kwargs,  # noqa
+    ):
+        name = name or re.sub(r"Model$", "", self.__class__.__name__)  # TODO: look name up from presets
+        super().__init__(
+            path=path,
+            freq=freq,
+            prediction_length=prediction_length,
+            name=name,
+            eval_metric=eval_metric,
+            hyperparameters=hyperparameters,
+            **kwargs,
+        )
+        self._lag_indices: List[int] = None
+        self._time_features: List[TimeFeature] = None
+        self._available_features: pd.Index = None
+        self._drop_median_prediction = False
+
+        # TODO: Should we always produce quantiles here?
+        if self.eval_metric == "mean_wQuantileLoss":
+            # Make sure that median is included in the quantiles - used for autoregressive generation
+            if 0.5 not in self.quantile_levels:
+                self._drop_median_prediction = True
+                self.quantile_levels = sorted(self.quantile_levels + [0.5])
+            self.tabular_predictor = TabularPredictor(
+                label=self.target,
+                problem_type=ag.constants.QUANTILE,
+                quantile_levels=self.quantile_levels,
+            )
+        else:
+            self.tabular_predictor = TabularPredictor(
+                label=self.target,
+                problem_type=ag.constants.REGRESSION,
+                eval_metric=TS_METRIC_TO_TABULAR_METRIC[eval_metric],
+            )
```

Comment:
As far as I understand, `problem_type=ag.constants.QUANTILE` is still somewhat experimental in AG-T, so some changes might happen to the API there. Training with `problem_type=ag.constants.REGRESSION` is much faster and more models are available for this task. However, the regression predictor cannot produce quantiles.

Here are the possible solutions that I see:
1. Always use `problem_type=QUANTILE`. 
    - Pros: Simpler code, less hassle for the user.
    - Cons: much longer training times and potentially worse performance for users that care about any metric other than `mean_wQuantileLoss`.
2. (Current code) Train with `problem_type=QUANTILE` if `eval_metric="mean_wQuantileLoss`, or `problem_type=REGRESSION` otherwise. 
    - Pros: Faster training, better accuracy of point forecast
    - Cons: not aligned with treating probabilistic forecasting as a first-class citizen
3. Fit two `TabularPredictor`s under the hood, one for 
    - Pros: Good point forecasts with quantiles available
    - Cons: "mean" and quantile forecasts may be incompatible, longer training times

Reply:
I did some profiling, we should be able to significantly speed up training / prediction if we vectorize `fit` and `predict` for the `BaseForestQuantileRegressor` in AG-T https://github.com/awslabs/autogluon/blob/bd1615584863e01b8123fb5d515e16cc5aca5639/tabular/src/autogluon/tabular/models/rf/rf_quantile.py#L347

Right now the training time for the quantile models is extremely long even for medium-sized datasets (e.g., `tourism_monthly`: 366 series with 60-300 timesteps each).

Reply:
I went with Option 1 for now and disabled the slow models with
```
excluded_model_types=["XT", "RF"]
```